src/vellum/workflows/nodes/displayable/bases/api_node/node.py

```diff
@@ -79,25 +85,36 @@ def _local_execute_api(self, data, headers, json, method, url):
             raise NodeException(f"Failed to prepare HTTP request: {e}", code=WorkflowErrorCode.PROVIDER_ERROR)
         try:
             with Session() as session:
-                response = session.send(prepped)
+                response = session.send(prepped, timeout=timeout)
         except RequestException as e:
             raise NodeException(f"HTTP request failed: {e}", code=WorkflowErrorCode.PROVIDER_ERROR)
         try:
-            json = response.json()
+            json_response = response.json()
         except JSONDecodeError:
-            json = None
+            json_response = None
         return self.Outputs(
-            json=json,
+            json=json_response,
             headers={header: value for header, value in response.headers.items()},
             status_code=response.status_code,
             text=response.text,
         )
 
-    def _vellum_execute_api(self, bearer_token, data, headers, method, url):
+    def _vellum_execute_api(self, bearer_token, data, headers, method, url, timeout):
         client_vellum_secret = ClientVellumSecret(name=bearer_token.name) if bearer_token else None
+
+        # Create request_options if timeout is specified
+        request_options = None
+        if timeout is not None:
+            request_options = RequestOptions(timeout_in_seconds=timeout)
```

Comment:
oh sick - I was worried that we'd have to pipe this through to Vellum's backend but this is better